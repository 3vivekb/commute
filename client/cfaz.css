/* CFAZ theme styling */

.navbar {
  background-color: #455a71;
}

.navbar-brand a {
  color: white;
}

.navbar-default .navbar-nav>li>a {
  color: #ddd;
}

.navbar-default .navbar-nav>li>a:hover {
  color: #fff;
}

footer {
  border: 0px;
  color: white;
  background-color: #455a71;
}

footer a {
  color: white;
}

.logo {
  width: 185px;
  height: 36px;
  margin-right: 15px;
  background-image: url("data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAfUAAABYCAMAAADbY56nAAAA8FBMVEVMaXH2pwT////0pAX3pwP////////1pgT1pwP////vnwCNwkn///+MwkiMwUf////1pgOMwkmMwkn////////1pgP////3pwf1pgSMwUj2pwP2pwT1pQP///////+Pv0////+Lw0f////2pwONwUmPv0ePxEqNwUj///////+NwUj3pwWNwUmMwkn////4vUP704L96cDj8NLG4aS/3Zn4+/T3rRTq9N2q0Xex1YKUxlTb7Mb++u/U6Lv4uDP5yGKizWv979DN5K+bymD83qG42Y385LHx9+j+9OD3siP72ZH5w1L6znL////2pwSNwkkXlKiSAAAAL3RSTlMA7xAwQL+Av4BAEJ/P72Bg34/fUCCfMCBwz4+vUK9wEH9A78+AIDCvn99wYL9Qj4T82U4AAA1TSURBVHhe7NfdrqIwFIbhj7SESjQSEgRBBfdmj//eySr3fzfTcUwwE38KwT1uWU+y0q6jHrxHxWtijDHGGGOM0ZmLgmgIwBwueoSrEwC3eXUxU6nXFP4r5lIBo6AhDQAaNqouEs/RbeA1cPW1uUxobV89VoFuC6+BqxcbFwsa21aXkTZ+enWuPifMPwZ21YWn9TtU5+oLmmzmVtX9VOv3qM7VB3SkhU31maPfpTpXB33QwKJ6qtvi6l9L4wvfypdyp9ROSv9qdZcIdfVusUO5D6uzcF8u8R2EuvxaO1kicHJ0xzDG7tjMERMznWOHVfWv0eqA54rTK1/rQMW4R+pGAs9LE/H3Ne9C5qPn8jKsrgrLHE8jM31DFNtVt+dkMwAy0LXPnmcvR9VNoy2eI4600aK71O04ygcSh7OfLKfVXdNfeILEedAo6bJ63T3+rPcIvVVWD23RNd/TD2V+x9WNQAJRvSr0U76vLKxydEoEVolE19WNFNjVm0Qf/Sbv7N/TxpE4bgw2iNCaXRtDeXGeNZfWLlvn2iZpk+z2dbvX3Tua//+/uRiwv7IYj62Y9C5Pvz+1EcSCj2Y0o5GUY3h3Vk8Oir3dqumR2wejDnX7Evah1tTuuXZk3SoJxA8AHdjvFzqPHdSbaSljn9YGYAaDNRRGzcH7VqmS/zF0YP+u0IG9scad3kzF3suxj2syt9eq4qbcxbpU1j1S//VGQ88OFchpQIcLbqj+qpCpG0Y+Drq1ALlhZuPWrTI2C/MBUv/pRkv/UPn5Qgh9AJoVcJ4KjHZm9JcS130TXrXklK3f0gnokp1Xd3eY/SDewBn4TamH39vD/0KhPXu11VuiTc3b0zFPouAxQf8s1XsQmtRyHK2xMa14T1/y83NjrmHs0Ya5bRaQbbiHfkPq1veO5MhF2PP1VtfU1H4A6mPZjteleglALW7mXSHz7lQHAxO8pA+fM64FfeDRPzYfBHV+deYyB3BF+fjm1Lv1qcN3lyubrYf4JxcBriQfP64bxgcp3YhocNKGxYOifkyuvV/nAM6oNfnjhtRhkBrUmZm3nZFbASigs9jH+Rpdi2fjAzo53YsHQJ039SsAeEMGdE2pd3Wp8zPvlDD1OQtgBGNv13uHxfjxQdr2AKjzpn62hj5Qxt6AOmxTgzo78wJ1z1jV9Nf9IYx9Wectbtodn8u9/IdD/fENpTdr6BP1gsfNqCvF1ZeKvubU/wRyBssYBLsI03h1EOev4Ckgypwj1hEkD4c6WV39sJZ0TpZhmlHnq6t//ZZB/3pRaGhVzNJL8O/VnmSWRh9mz8/qXmmzo7p404ms7WpO5JSG96YQniZ1Twj8OjfZPGOARyhdGKTNVuIqLSc3lH5fy7qksB9XU/eCRfrUcP+pcxb6xccM+m/vlKYO7zqmxoQ0dd7Y4eLn/PpMxMBICUtsojVUzPCTbOnFS9JM365B3bIsZ9MgolAKG52YeASY2yGaY6fKwSNZv0bKvqcXOHQqSZR88NjR2Lv85zrTqdrU4822Y8w0iinDPO6fVjuIlBBMpta6LYQoEKdy7bShHvXdy8yF/CWbFqASy4M+hsT+ovERRfTLLnr/wKTsP1VQt9dMp5Yc9L9z6O/rLstmzX1jyDsFcuyNct8zYx18WBe6s8Fg2a4QIohiYAd1c7DWpe6HMK3sf2HkCOHaWB4sdiFOXCGcJEQPmGn9OsvU/1Wesv/KUjejLWpbCOFE2w4KiRKj9zn0P4hWlvoQM7TOmkE3Tym6LMgFGqoT+wVigCDcYgP1HfTQsm1Rk/oOc7ywbS+19A3J3K27cY4dOUXo5CYYFrFT0D/vEH6+OWNS9t2vTzVIeW+U9WLjiywPn7uQ2bQZ6O+opA1q8/gAso4wRCrXaRKAq5SFEAAMQlOmnqScXbRXU9/mEDlXSzFus7BiYMqUM8+AHOMpxfN1Pp1/ZlL2Yzqag3sPDMjbhHUeKJVCz8P3jxdUe4ejPjM6mNY1JnbwZ0mK+qZu7VdtHIm6uEUWGIYOdbtQx3cAXcYe4HmADuw+U27L/PoX5O1Uyv4LSR0uzjGUTuGT9XTDd2jFUe/lIXxPqxaACgyTrdembiPuk7EmEnULyGpST0JgpGNLL8wjDw8mVlhkWjDU3yKGy1z8uRZ1uDh1sIkq6i/p8B3q8dR7d6Le4agjftFw8AQ3C80xRhC/qwLU490UAVO3qNHm5KZuU93yQL0kWf9dnuIvdahjGlM7ZVVQ/0MK3x8s9RBEaOow/NrUFdNeUJ7HDDNrDvH9F6kEpdSvznPQcPHXNanTQw0RhgfqNcJ3feorLepgvTwgddu2nSrqpjb1gdKbmC7/Z3wjcsXBAnU6WT8vlmGuNKineE06DA446qfK6rs2dcTwo2bRHE9dX26ROpiAemwTkqg7aphAZ5eCCixgjKXUX8HBSy7+rA519GnBxrY93fAdmhwwc+sjX8O/7oO6OVCoO3rVF7UEENBY/Z1ZLVSrw7t8mjowv93qTVnKflJGPdg8nFAeZM4rwve/iGZ2zQ3QdFZp5sxg0Y3hmfqKHYVrhbqvTT1UQySzfOXWomv9Ah9hD+anNSEqZSeiVvRJsMFthw/fkbPtq82W8Ma5055rrMhO83Bg1jhfh0SQWIM1JFM3tKlbamcsQrvXley4HSDc+plI1iGmyv6EwAm4ZG0xye1FN3yH+LBsbow0JvYWU33RXpvzhBA+apyF0kjoHJo6LVAnBerPiF2SEFNlP2KpGxDlBIZM+P43B31ZabUoxFRppVOzCarW4W1pWEg1ztiyHd8w/s+oP+cq61yV/XET6iOV5r+rwne+gDpH+aX+rorh/laM/t1rbotdNINymmULD9wOTd2m5ezyOpuWIBfiryjeZMp+0oT6Sjt8Z6drsG7jBNO4rqmvjEnOv6q+zrf7sEQLYfohqWOEVS0T8fqZ3CX5WtIrKmV/QoVp6JNXdojAByXo4mt1+M5nVst8Ou8QpXj+sAwcPFeziZBm877AQeXpvqjbJd8wcMRaG6OpPO2SStmfGw1i+O2ZQujjOtN/TlXB9rkYbSIhRGrPqYupYM4lCCDDftVJPu8PkI/fF3WH9zsREU5zG+c+ALCkc2IoHJdTd8vCXakCPCfW5Gid1joGMZYZwnWXa0QceRoanGLO2M0wA2HiU0LeYan7JRty0zSifBHHFAKBxhGRrF+REd4HNYKnqXugW5Ar+8jhXagvOdMFRGBnoBPn4nrV26J8JoKPy1dL3cNSN0J1ZOHRGBRsBfiE2CVJZ3OfiFgO1KEB7Qsj+akrXeo8RfnwQ7vyGGx/JA8k6TjEXc+++MiKBOXpFgejDh8uyuOmmKi5bX8KGkd7eL/QW2bPpa2SHPWA9IVeMfdZalPn47MhJudJxfVV7aW81tdD0HCHc27I1WLZ4NR3NqfOvQE+NsbaAeWrIurM0zUcPOniL3G4kaBO1VTL668dfeodPhHDtSMz9ra6eRdawTPA1PkzrSR02J5JVEEHh6aOEg69YhDuH6g3kVoqKzWvtnq9t7tGaXhRsXvEJmZ2V91qMdWlPuWZDGGz/WXhRkEGJ14KU9c5vw7oiRz1BcrbwkFj6sy5OthybAJAbCqdQCUUy7IaOmI252BcRfsbqAL6eOHFKaOLmtd+yrH7liU0HE06lPppCokwUG6qf1eFExZdgJ1vZMRGUWFhyDeljkwxdFRH5MreJfYlIhG20mleQYVLqIguOPs7wk3lMMiCu4Kq+SVUMwI7q5Ecy0+K2aTmvTRRcchjp7qf2liwgRTbiX8Q6pg1FhlYoezL8rbb5T2cjSKyuaeP6kN/dEImNaGb0jblNCd0so++IO/wWOlQry6djlvAXuem0qkMvbvxPZDeHVS24tZSWbZtL+LdkHBxSOQQ1HH8aRDtHqMEkf5g24nkttkKcSbijtgfPSXXKNQTTwIfPdqZA6BrYAfHas0LL+9V+I45oOMOo0rqODQIWb5RcsQM33Z0UOo4WgKFbpFJwpw4A3Z96JAD6hiLXJ+AvTl0aFQw5M6Qv5FYngXm6k6PRndLmkHOPbbN/NDxwva2Z1pVhH7FFWMlzaYdA+r+WWYvwrBYuAapp7Xm9icnBikvsXbUIR9PRZ+Yu99Yu6ynwkVzzCXk3U7xD0H19q48rHmPrC3KXiHsW7m+cX/CYxxh0s2+u+mFYK4qeXbQK6Px1ECYzS6NXraNmpLNt9XDTZJFzW6Zj2doQK4HGT+Mnld4+UcvjEMKq6O8pv3aAIr0hqt0WE2LQLuTsWH0ey01lv9hqRsnR2yafmzcizrDiuv7ddSfdSWN+tsnTDZ/i7vXW228RnvUUhK4eVfVf9u7uxSEYSgIoyMGDBVLFEqCFST0R/C6Ft3/bsQ+KTZowStK5mzhe8rDzSAnfbL7soMaSXZ3gg8ztnl6tVO33owtPHVQJfV5RL17sfk3TbDVInVWQX17Wt2PuR176Cuknj0ucMlbf4elTd4WoX2MZduWMe7xPd4EOwjGY6BcvSrwU0i/ujPIDKs7QWZYvRHkhdXddo6csPow8f2/yNupxBTQQURERERERERXSpSiTPMvm1QAAAAASUVORK5CYII=")
  background-position: 0px 0px;
  background-size: auto 32px;
  background-repeat: no-repeat;
}

.welcome-image {
  background-image: url(https://raw.githubusercontent.com/conveyal/modeify/dev/images/application/welcome-background.jpg);
  background-repeat: no-repeat;
  background-size: cover;
}

.table-bordered > thead {
  background-color: #778595;
  color: white;
}

.table-bordered > thead > tr > th {
  border: 1px solid #778595;
}
