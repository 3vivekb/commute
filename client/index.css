
@import url(node_modules/normalize.css/normalize.css);
@import url(node_modules/font-awesome/css/font-awesome.css);
@import url(node_modules/react-bootstrap-table/dist/react-bootstrap-table.min.css);
@import url(node_modules/react-select/dist/react-select.css);
@import url(https://maxcdn.bootstrapcdn.com/bootstrap/latest/css/bootstrap.min.css);

footer {
  border-top: 1px solid gray;
  margin-top: 3em;
  padding: 1em;
}
