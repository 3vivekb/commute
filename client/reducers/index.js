import {handleActions} from 'redux-actions'

import * as commuter from './commuter'
import * as multiSite from './multi-site'
import * as site from './site'
import * as user from './user'

export default {
  commuter: handleActions(commuter.reducers, commuter.initialState),
  multiSite: handleActions(multiSite.reducers, multiSite.initialState),
  site: handleActions(site.reducers, site.initialState),
  user: handleActions(user.reducers, user.initialState)
}
