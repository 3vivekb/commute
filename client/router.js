import React from 'react'
import {IndexRoute, Router, Route} from 'react-router'

import AddCommuters from './containers/add-commuters'
import Application from './containers/application'
import EditCommuter from './containers/edit-commuter'
import EditMultiSite from './containers/edit-multi-site'
import EditSite from './containers/edit-site'
import MultiSite from './containers/multi-site'
import MultiSiteCreateReport from './containers/multi-site-create-report'
import MultiSiteReport from './containers/multi-site-report'
import Site from './containers/site'
import SiteCreateReport from './containers/site-create-report'
import SiteReport from './containers/site-report'
import UserHome from './containers/user-home'

const ApplicationRouter = ({history}) => (
  <Router history={history}>
    <Route path='/' component={Application}>
      <IndexRoute component={UserHome} />
      // Site Views
      <Route path='/site/create' component={EditSite} />
      <Route path='/site/:siteId/edit' component={EditSite} />
      <Route path='/site/:siteId' component={Site} />
      <Route path='/site/:siteId/bulk-add-commuters' component={AddCommuters} />
      <Route path='/site/:siteId/commuter/create' component={EditCommuter} />
      <Route path='/site/:siteId/commuter/:commuterId/edit' component={EditCommuter} />
      <Route path='/site/:siteId/create-report' component={SiteCreateReport} />
      <Route path='/site/:siteId/report' component={SiteReport} />

      // Multi-Site Views
      <Route path='/multi-site/create' component={EditMultiSite} />
      <Route path='/multi-site/:multiSiteId/edit' component={EditMultiSite} />
      <Route path='/multi-site/:multiSiteId' component={MultiSite} />
      <Route path='/multi-site/:multiSiteId/create-report' component={MultiSiteCreateReport} />
      <Route path='/multi-site/:multiSiteId/report' component={MultiSiteReport} />
    </Route>
  </Router>
)

export default ApplicationRouter
