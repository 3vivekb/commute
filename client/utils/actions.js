import fetchAction from '@conveyal/woonerf/fetch'
import qs from 'qs'
import {push} from 'react-router-redux'
import {createAction} from 'redux-actions'

/**
 * Make generic model actions that communicate with a rest server
 *
 * @param  {Object} cfg An object with the following parameters:
 * - commands: An object with the keys being commands and the corresponding cfg of each command
 * -- redirectionStrategy: The redirection strategy to use upon success.
 *  Can be one of following: 'toEntity', 'toParent', 'toHome', undefined (no redirection)
 *  Defaults are as follows:
 *    Collection GET, GET: null
 *    Collection POST, PUT: toEntity
 *    DELETE: toParent
 * - parentKey: Key in model of parent's id
 *   (must be provided if using toParent redirection strategy)
 * - parentName: Name of app route to parent view
 *   (must be provided if using toParent redirection strategy)
 * - pluralName: Plural name of the model
 * - singularName: Singular name of the model
 * @return {Object}     The actions
 */
const makeGenericModelActions = (cfg) => {
  const {commands, parentKey, parentName, pluralName, singularName} = cfg
  const actions = {}

  // make local set actions
  const addLocally = createAction(`add ${singularName}`)
  const deleteLocally = createAction(`delete ${singularName}`)
  const setLocally = createAction(`set ${singularName}`)
  const setAllLocally = createAction(`set ${pluralName}`)

  const baseEndpoint = `/api/${singularName}`

  const redirectionStrategies = {
    'toEntity': (entity) => {
      return push(`/${singularName}/${entity._id}`)
    },
    'toHome': () => {
      return push('/')
    },
    'toParent': (entity) => {
      return push(`/${parentName}/${entity[parentKey]}`)
    }
  }

  /**
   * Do a redirect if needed
   *
   * @param  {Object} cfg Paremters as follows:
   *   - actions: The actions to append to
   *   - endpointCfg: The endpoint config
   *   - defaultStrategy: The default redire strategy to use (default: undefined)
   *   - redirectArgs: Arguments to provide to the redirectionStrategy
   */
  const doRedirectIfNecessary = (cfg) => {
    const redirectionStrategy = (cfg.endpointCfg.redirectionStrategy !== undefined
      ? cfg.endpointCfg.redirectionStrategy
      : cfg.defaultStrategy)

    if (redirectionStrategy) {
      const redirect = redirectionStrategies[redirectionStrategy](cfg.redirectArgs)
      if (redirect) {
        cfg.actions.push(redirect)
      }
    }
  }

  if (commands['Collection GET']) {
    actions.loadAll = (queryParams) => {
      // only include filteredKeys in querystring
      let queryString = ''
      if (queryParams) {
        queryString = qs.stringify(queryParams)
        if (queryString.length !== 0) {
          queryString = '?' + queryString
        }
      }

      // make request
      return fetchAction({
        next: (res, err) => {
          if (!err) {
            return setAllLocally(res.value)
          }
          // TODO handle error
        },
        url: baseEndpoint + queryString
      })
    }
  }

  if (commands['Collection POST']) {
    const endpointCfg = commands['Collection POST']
    actions.create = (newEntity) => {
      return fetchAction({
        next: (res, err) => {
          if (err) {
            // TODO handle error
          } else {
            const createdEntity = res.value
            const actions = [
              addLocally(createdEntity)
            ]
            doRedirectIfNecessary({
              actions,
              endpointCfg,
              defaultStrategy: 'toEntity',
              redirectArgs: createdEntity
            })
            return actions
          }
        },
        options: {
          body: newEntity,
          method: 'POST'
        },
        url: baseEndpoint
      })
    }
  }

  if (commands['DELETE']) {
    const endpointCfg = commands['DELETE']
    actions.delete = (entity) => {
      return fetchAction({
        next: (res, err) => {
          if (err) {
            // TODO handle error
          } else {
            const actions = [
              deleteLocally(entity._id)
            ]
            doRedirectIfNecessary({
              actions,
              endpointCfg,
              defaultStrategy: 'toParent',
              redirectArgs: entity
            })
            return actions
          }
        },
        options: {
          method: 'DELETE'
        },
        url: `${baseEndpoint}/${entity._id}`
      })
    }
  }

  if (commands['GET']) {
    actions.loadOne = (id) => fetchAction({
      next: (res, err) => {
        if (!err) {
          return setLocally(res.value)
        }
        // TODO handle error
      },
      url: `${baseEndpoint}/${id}`
    })
  }

  if (commands['PUT']) {
    const endpointCfg = commands['PUT']
    actions.update = (entity) => fetchAction({
      next: (res, err) => {
        if (err) {
          // TODO handle error
        } else {
          const updatedEntity = res.value
          const actions = [
            setLocally(updatedEntity)
          ]
          doRedirectIfNecessary({
            actions,
            endpointCfg,
            defaultStrategy: 'toEntity',
            redirectArgs: updatedEntity
          })
          return actions
        }
      },
      options: {
        body: entity,
        method: 'PUT'
      },
      url: `${baseEndpoint}/${entity._id}`
    })
  }

  return actions
}

export default makeGenericModelActions
