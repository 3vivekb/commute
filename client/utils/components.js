/**
 * Humanize distances
 * // TODO: more humanization
 *
 * @param  {float} d Distance to humanize
 * @return {string} humanized distance
 */
export function humanizeDistance (d) {
  return `${d} miles`
}
