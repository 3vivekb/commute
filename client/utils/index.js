import currencyFormatter from 'currency-formatter'
import yup from 'yup'

export function actUponConfirmation (confirmationMessage, action) {
  if (window.confirm(confirmationMessage)) {
    action()
  }
}

export function arrayCountRenderer (cell, row) {
  if (Array.isArray(cell)) return cell.length
}

/**
 * Calculate number of elements in array less than or equal to target values
 *
 * @param  {Number[]} arr     A sorted array of numbers
 * @param  {[Number]} target  The number to be less than or equal to
 * @return {int}              The number of elements in the array less than or equal to the target value
 */
export function calcNumLessThan (arr, target) {
  // do binary search
  let left = 0
  let right = arr.length - 1
  let mid

  while (left <= right) {
    mid = Math.floor((left + right) / 2)
    if (arr[mid] <= target) {
      left = mid + 1
    } else {
      right = mid - 1
    }
  }

  return left
}

/**
 * Round a number to a fixed amount of decimal places
 *
 * @param  {Number} n      The number to round
 * @param  {Number} places The number of places to round to (default=2)
 * @return {Number}        The rounded number
 */
export function fixedRound (n, places) {
  places = places || 2
  const multiplier = Math.pow(10, places)
  return Math.round(n * multiplier) / multiplier
}

export function formatCurrency (n) {
  return currencyFormatter.format(n, { code: 'USD' })
}

export const geocodeResultToState = {
  address: result => result ? result.properties.label : undefined,
  city: result => result ? result.properties.locality : undefined,
  coordinate: result => {
    return result ? {
      lat: result.geometry.coordinates[1],
      lon: result.geometry.coordinates[0]
    } : undefined
  },
  country: result => result ? result.properties.country : undefined,
  county: result => result ? result.properties.county : undefined,
  geocodeConfidence: result => result ? result.properties.confidence : undefined,
  neighborhood: result => result ? result.properties.neighborhood : undefined,
  state: result => result ? result.properties.region : undefined
}

export const geocodeYupSchema = {
  address: yup.string().label('Address').required(),
  city: yup.string(),
  coordinate: yup.object({
    lon: yup.number().required(),
    lat: yup.number().required()
  }),
  country: yup.string(),
  county: yup.string(),
  geocodeConfidence: yup.number(),
  neighborhood: yup.string(),
  original_address: yup.string(),
  state: yup.string().required()
}

/**
 * Humanize distances
 * // TODO: more humanization
 *
 * @param  {float} d    Distance to humanize
 * @param  {int} places number of decimal places to round to
 * @return {string}     humanized distance
 */
export function humanizeDistance (d, places) {
  places = places || 1
  return `${fixedRound(d, places)} miles`
}
