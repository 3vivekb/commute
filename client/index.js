
import mount from 'mastarm/react/mount'
import createStore from 'mastarm/react/store'

import reducers from './reducers'
import router from './router'

mount({
  id: 'root',
  router,
  store: createStore(reducers)
})
