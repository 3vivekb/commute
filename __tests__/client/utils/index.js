/* globals describe, expect, it */

import * as util from '../../../client/utils'

describe('utils', () => {
  it('fixedRound should work', () => {
    expect(util.fixedRound(1.23456)).toEqual(1.23)
  })

  it('humanizeDistance should work', () => {
    expect(util.humanizeDistance(123)).toMatchSnapshot()
  })

  describe('> calcNumLessThan', () => {
    const testCases = [{
      arr: [],
      target: 3,
      expected: 0
    }, {
      arr: [1],
      target: 3,
      expected: 1
    }, {
      arr: [4],
      target: 3,
      expected: 0
    }, {
      arr: [1, 2, 3, 4],
      target: 0,
      expected: 0
    }, {
      arr: [1, 2, 3, 4],
      target: 5,
      expected: 4
    }, {
      arr: [1, 2, 3, 4],
      target: 2,
      expected: 2
    }, {
      arr: [1, 2, 3, 4],
      target: 3,
      expected: 3
    }, {
      arr: [1, 2, 3],
      target: 2,
      expected: 2
    }, {
      arr: [1, 1, 1, 1],
      target: 1,
      expected: 4
    }, {
      arr: [1, 1, 1, 1],
      target: 0,
      expected: 0
    }, {
      arr: [1, 2, 2, 2, 5],
      target: 2,
      expected: 4
    }]

    testCases.forEach((tc, idx) => {
      it(`should compute correct amount for test case ${idx}`, () => {
        expect(util.calcNumLessThan(tc.arr, tc.target)).toBe(tc.expected)
      })
    })
  })
})
