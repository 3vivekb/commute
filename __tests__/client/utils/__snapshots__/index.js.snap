exports[`utils humanizeDistance should work 1`] = `"123 miles"`;
